File: app/src/main/java/com/list_sample/preferencesample/PreferenceDemoFragment.java
package com.list_sample.preferencesample;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.preference.PreferenceFragmentCompat;

/**
 * Created by monkey on 2017/10/25.
 */

public class PreferenceDemoFragment extends PreferenceFragmentCompat {
    private OnFragmentInteractionListener mListener;

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.preferences, rootKey);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface OnFragmentInteractionListener {}
}

File: app/src/main/java/com/list_sample/preferencesample/PreferenceDemoFragment.kt
package com.list_sample.preferencesample

import android.content.Context
import android.os.Bundle
import android.support.v14.preference.PreferenceFragment
import android.support.v7.preference.PreferenceFragmentCompat
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by monkey on 2017/10/25.
 */

class PreferenceDemoFragment : PreferenceFragmentCompat() {

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        activity.showPreferenceButton.visibility = 4

        addPreferencesFromResource(R.xml.preferences)
    }

    override fun onDetach() {
        super.onDetach()
        activity.showPreferenceButton.visibility = 0
    }
}
